quil-rs/src/instruction/gate.rs

```diff
@@ -187,24 +257,31 @@ impl GateDefinition {
 
 impl fmt::Display for GateDefinition {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        let parameter_str = get_string_parameter_string(&self.parameters);
-        writeln!(
+        write!(
             f,
-            "DEFGATE {}{} AS {}:",
+            "DEFGATE {}{}",
             self.name,
-            parameter_str,
-            match self.specification {
-                GateSpecification::Matrix(_) => "MATRIX",
-                GateSpecification::Permutation(_) => "PERMUTATION",
-            }
+            get_string_parameter_string(&self.parameters)
         )?;
+        match &self.specification {
+            GateSpecification::Matrix(_) => writeln!(f, " AS MATRIX:")?,
+            GateSpecification::Permutation(_) => writeln!(f, " AS PERMUTATION:")?,
+            GateSpecification::PauliSum(sum) => writeln!(
+                f,
+                " {} AS PAULI-SUM:",
+                sum.get_arguments()
+                    .into_iter()
+                    .collect::<Vec<String>>()
+                    .join(" ")
+            )?,
```

Comment:
Would prefer to avoid string allocation inside of `Display`.

```suggestion
            GateSpecification::PauliSum(sum) => {
                for arg in sum.get_arguments() {
                    write!(f, " {arg}")?;
                }
                writeln!(f, " AS PAULI-SUM:")?;
            )?,
```